### ISSUE TITLE
dry activation scripts break with dependencies not supporting dry activation

### ISSUE THREAD

**Issue by pennae** [NixOS/nixpkgs#137304]
### Describe the bug

adding a dependency to the `users` activation script that does not support dry activation makes the system unbuildable. making the dependency itself dry-runnable may not be sufficient, since eg `specialfs` is not dry-runnable so adding *that* as a dependency will also break things.

setting dependencies for `users` specifically only breaks after a851b4d20ee0dad33395f85a092df47ac9b1cc3e, which added dry-run support for that activation script.

### Steps To Reproduce
add this to configuration.nix and rebuild:
```
system.activationScripts = {
  users.deps = [ "x" ];
  x.text = "true";
};
```

### Expected behavior
system builds

### Screenshots
```
building Nix...
building the system configuration...
error: attribute 'x' missing, at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/strings-with-deps.nix:72:71
```

### Notify maintainers
@dasJ 

### Metadata
```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.14.1, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.11`
 - channels(root): `"nixos-21.11pre313614.09cd65b33c5"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by veehaitch** [comment]
ref: https://github.com/NixOS/nixpkgs/pull/136605

**Comment by pennae** [comment]
fixed, presumably by #137508
